Sources/Commands/Utilities/TestingSupport.swift

```diff
@@ -156,6 +159,10 @@ enum TestingSupport {
             env["NO_COLOR"] = "1"
         }
 
+        // Set an environment variable to indicate which library's test product
+        // is being executed.
+        env["SWIFT_PM_TEST_LIBRARY"] = String(describing: library)
```

Comment:
This is convenient for shared libraries which want to have support code that might be run by either an XCTest or SwiftTesting test case.

However, I suggest we name this more generally (for example, `APPLE_TEST_LIBRARY`) because it's not SwifPM-specific, and other build tools / IDEs might want to have a uniform way to identify the Apple-provided testing library in use.

I also expect we _wouldn't_ this environment variable if SwiftPM gained direct support for additional testing libraries from other platforms/environments, since those would likely be run via their own test harnesses which might define their own environment variables.